### PR TITLE
backend/ondc-interoperability: fixes confirm and on_confirm according to new spec

### PR DESCRIPTION
**Pull request by jaypalmudaliyar24**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

makes confirm & on_confirm interoperable.

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/Confirm.hs [hunk 1]
```diff
@@ -236,7 +81,7 @@ tfFulfillments res =
       DRB.OneWaySpecialZoneDetails _ -> "RIDE_OTP"
       DRB.RentalDetails _ -> "RENTAL"
       DRB.InterCityDetails _ -> "INTER_CITY"
-      _ -> "RIDE"
+      _ -> "DELIVERY"
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/Confirm.hs:84]: use enum here

**Review by NikithShetty — CHANGES_REQUESTED** [review]

File: Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/Confirm.hs (hunk 1 above)
Comment (jaypalmudaliyar24) [reply Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/Confirm.hs]: Fixed.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs [hunk 2]
```diff
@@ -232,13 +71,14 @@ tfFulfillments res = do
...
-    mkFulfillmentState quoteType =
+    mkFulfillmentState quoteType = do
+      fulfilState <- bookingStatusCode quoteType
       Just $
         Spec.FulfillmentState
           { fulfillmentStateDescriptor =
               Just $
                 Spec.Descriptor
-                  { descriptorCode = Just (bookingStatusCode quoteType),
+                  { descriptorCode = Just fulfilState,
```
Comment (RitikaHanish) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs:81]: You are sending Nothing and putting Just also.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs [hunk 3]
```diff
@@ -248,7 +88,7 @@ tfItems :: DConfirm.DConfirmResp -> Maybe [Spec.Item]
 tfItems res =
   Just
     [ Spec.Item
-        { itemDescriptor = tfItemDescriptor res,
+        { itemDescriptor = Nothing,
```
Comment (RitikaHanish) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs:91]: Why is this removed ?

File: Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/Confirm.hs [hunk 4]
```diff
@@ -310,6 +155,7 @@ tfCustomer res =
       Just $
         Spec.Contact
           { contactPhone = Just res.riderPhoneNumber
+          -- handling of passing virtual number at on_init domain handler.
```
Comment (RitikaHanish) [inline Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/Confirm.hs:158]: is it done ?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs [hunk 5]
```diff
@@ -12,183 +12,21 @@
...
-            },
-        agent = agent
-      }
-
-bookingStatusCode :: DConfirm.ValidatedQuote -> Text
-bookingStatusCode (DConfirm.DriverQuote _ _) = "TRIP_ASSIGNED"
-bookingStatusCode (DConfirm.StaticQuote _) = "NEW"
-bookingStatusCode (DConfirm.RideOtpQuote _) = "NEW"
+bookingStatusCode :: DConfirm.ValidatedQuote -> Maybe Text
+bookingStatusCode (DConfirm.DriverQuote _ _) = Nothing
```
[the remark this answers is not in the document]
Comment (jaypalmudaliyar24) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs:27]: As we discussed, we can't send `RIDE_ASSIGNED`.
So kept it `Nothing` & will use `on_update` to send `RIDE_ASSIGNED` event.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs (hunk 2 above)
Comment (jaypalmudaliyar24) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs:81]: No if you check above part I have used a bind operator so if the `fulfillState` is `Nothing`. It will send `Nothing` for whole `FulFillmenState` as `Nothing`. else it will use it in the below way.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs (hunk 3 above)
Comment (jaypalmudaliyar24) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs:91]: It was not consumed at BAP & it's value is as same as `itemId`.

File: Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/Confirm.hs (hunk 4 above)
Comment (jaypalmudaliyar24) [reply Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/Confirm.hs:158]: Yes this is fixed.

File: Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/OnConfirm.hs [hunk 6]
```diff
@@ -12,89 +12,73 @@
...
 
-      return (bppBookingId, startOtp)
+      let fulfState = fulf >>= (.fulfillmentState) >>= (.fulfillmentStateDescriptor) >>= (.descriptorCode)
+      let isRideAssigned = fulfState == Just (show Enums.RIDE_ASSIGNED)
+
+      if isRideAssigned
+        then do
+          let driverImage = fulf >>= (.fulfillmentAgent) >>= (.agentPerson) >>= (.personImage) >>= (.imageUrl)
+              driverMobileCountryCode = Just "+91" -- TODO: check how to get countrycode via ONDC
+              driverRating = Just $ toCentesimal 5 -- Default value for driver_rating for not-value-add-np.
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/OnConfirm.hs:61]: Move this default to UI API.

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/Common.hs [hunk 7]
```diff
@@ -123,20 +123,17 @@ rideAssignedReqHandler ::
     MonadFlow m,
     EncFlow m r,
     EsqDBReplicaFlow m r,
-    HasHttpClientOptions r c,
+    -- HasHttpClientOptions r c,
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/Common.hs:126]: Remove unused comments

**Review by NikithShetty — APPROVED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs (hunk 3 above)
Comment (NikithShetty) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs:91]: This will be added in next change

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/Common.hs (hunk 7 above)
Comment (jaypalmudaliyar24) [reply Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/Common.hs]: Fixed.
